tools/add_fib.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python
+
+# CMSIS-DAP Interface Firmware
+# Copyright (c) 2009-2013 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from __future__ import absolute_import
+from __future__ import print_function
+
+import argparse
+import itertools
+import binascii
+import intelhex
+import os
+
+FIB_BASE = 0x2000
+FLASH_BASE = 0x3000
+FW_REV = 0x01000100  #todo: determine if revision can be passed from yotta
+
+def ranges(i):
+    for _, b in itertools.groupby(enumerate(i), lambda x_y: x_y[1] - x_y[0]):
+        b = list(b)
+        yield b[0][1], b[-1][1]
+
+
+def add_fib_at_start(arginput):
+    #parser = argparse.ArgumentParser(description='Firmware Information Block generation script')
+    #parser.add_argument("input", type=str, help="bin file to read from.")
+
+    #args = parser.parse_args()
+    #input_file = args.input + "_orig.bin"
+    #output_file = args.input #use same name, does not include extension
+    #file_name_hex = args.input + ".hex"
+    #file_name_bin = args.input + ".bin"
+
+    print("inputfile", arginput)
+    input_file = arginput + ".bin"
+    output_file = arginput #use same name, does not include extension
+    file_name_hex = arginput + "_fib.hex"
+    file_name_bin = arginput + ".bin"
+    print("inputfile", input_file)
+    print("output_file", output_file)
+    print("file_name_hex", file_name_hex)
+    print("file_name_bin", file_name_bin)
+
+
+    # Import intelhex if avaialable, otherwise fail
+    try:
+        from intelhex import IntelHex
```

Comment:
This is redundant, intelhex is already guaranteed to exist from the prior import.